mypy/checkmember.py

```diff
@@ -107,11 +107,17 @@ def analyze_member_access(name: str,
     elif isinstance(typ, UnionType):
         # The base object has dynamic type.
         msg.disable_type_names += 1
+        old_num_messages = msg.num_messages()
         results = [analyze_member_access(name, subtype, node, is_lvalue, is_super,
                                          is_operator, builtin_type, not_ready_callback, msg,
                                          original_type=original_type, chk=chk)
                    for subtype in typ.items]
         msg.disable_type_names -= 1
+        if msg.num_messages() != old_num_messages and any(isinstance(t, AnyType)
+                                                          for t in results):
+            # If there was an error, return AnyType to avoid generating multiple messages for the
+            # same error.
```

Comment:
This fixes a regression in `testUnionAndBinaryOperation` (and another test case whose name I can't remember right now). Previously, if some of the results was `Any`, the result got simplified into `Any`. Without this change, we'd preserve the non-`Any` types and this would sometimes produce a duplicate, similar-looking error message about the same binary operation, which would be a regression. Now the code falls back to the old behavior in these cases, and since this only happens when we have already generated a message, it doesn't affect correctness, just the specific messages generated for type errors involving union types.